addons/necropost-highlighter/addon.json

```diff
@@ -0,0 +1,198 @@
+{
+	"name": "Necropost highlighter",
+	"description": "Spot old forum topics brought back to life, before opening the message. Topics started much earlier than others listed on the same page are highlighted (in selected forums). If you've wasted time reading zombie messages or a help/question/request that was actually asked years ago, this is for you!",
+	"info": [
+		{
+			"type": "notice",
+			"id": "whatGetsHighlighted",
+			"text": "Sometimes an older topic is kept alive for good reasons, like an ongoing conversation, or the original poster following up their own topic months later. All older topics in selected forums are highlighted, including occasional lingering topics that are not necroposts. (That's why they get highlighted, but not removed.)"
+		}
+	],
+	"credits": [
+		{
+			"name": "RT_Borg",
+			"link": "https://github.com/RT-Borg"
+		}
+	],
+	"dynamicDisable": true,
+	"dynamicEnable": true,
+	"enabledByDefault": false,
+	"tags": [
+		"community",
+		"forums"
+	],
+	"versionAdded": "1.28.0",
+	"userscripts": [
+		{
+			"url": "userscript.js",
+			"matches": [
+				"https://scratch.mit.edu/discuss/search/*",
+				"https://scratch.mit.edu/discuss/6/*",
+				"https://scratch.mit.edu/discuss/m/6/*",
+				"https://scratch.mit.edu/discuss/7/*",
+				"https://scratch.mit.edu/discuss/m/7/*",
+				"https://scratch.mit.edu/discuss/9/*",
+				"https://scratch.mit.edu/discuss/m/9/*",
+				"https://scratch.mit.edu/discuss/11/*",
+				"https://scratch.mit.edu/discuss/m/11/*",
+				"https://scratch.mit.edu/discuss/4/*",
+				"https://scratch.mit.edu/discuss/m/4/*",
+				"https://scratch.mit.edu/discuss/1/*",
+				"https://scratch.mit.edu/discuss/m/1/*",
+				"https://scratch.mit.edu/discuss/3/*",
+				"https://scratch.mit.edu/discuss/m/3/*",
+				"https://scratch.mit.edu/discuss/49/*",
+				"https://scratch.mit.edu/discuss/m/49/*"
```

Comment:
```suggestion
				"https://scratch.mit.edu/discuss/*"
```

Reply:
This addon only runs in specific forums so a regex match could be used.

Reply:
The addon really shouldn't run on most discuss pages. Not on the top level directory, not on any of the actual message threads, and not on the many forums where necroposts aren't really an issue (at least not one that can be solved like this, because topics are expected to continue over long periods of time.)

Is there a heavy cost to matching accurately?

Reply:
You can use regex if you want - just prefix it with `^` like any other head-matching regexes.

Reply:
Thank you. Unless someone has a magical incantation they can share with me that gets the job done, I'll look up "head-matching regexes" tomorrow.

Reply:
After some false starts and learning json doesn't escape regex special characters in strings, this seems to be the magic:
"^https://scratch.mit.edu/discuss/(?:m/)?(?:search|1|3|4|6|7|9|11|49)/.*"
Lightly tested and pushed, but can someone with deep regex fu please verify.

Reply:
@Weredime suggested an improvement in review. Thank you.